channels/channelmapper.go

```diff
@@ -60,79 +113,149 @@ func GetDefaultSyncFunction(scopeName, collectionName string) string {
 
 }
 
-func (mapper *ChannelMapper) MapToChannelsAndAccess(body map[string]interface{}, oldBodyJSON string, metaMap map[string]interface{}, userCtx map[string]interface{}) (*ChannelMapperOutput, error) {
-	numberFixBody := ConvertJSONNumbers(body)
-	numberFixMetaMap := ConvertJSONNumbers(metaMap)
+func (mapper *ChannelMapper) Function() string {
+	return mapper.fnSource
+}
+
+// This function is DEPRECATED. It's currently used in some tests that can't easily be changed.
+// Its current implementation is a kludge, and it shouldn't be used in production.
+func (mapper *ChannelMapper) SetFunction(fnSource string) error {
+	host := mapper.service.Host()
+	mapper.fnSource = fnSource
+	mapper.service = js.NewService(host, "channelMapper", wrappedFuncSource(fnSource, host))
+	return nil
+}
+
+func wrappedFuncSource(funcSource string, host js.ServiceHost) string {
+	// Use different wrapper scripts for modern vs older JS.
```

Comment:
Is the difference here going to be otto vs v8, or is this within versions of v8?

Reply:
Otto vs V8. Otto is stuck at a very old version of JS.